metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/queries_extractor.py

```diff
@@ -276,18 +276,23 @@ def get_workunits_internal(
             logger.info(f"Found {self.report.num_unique_queries} unique queries")
 
         with self.report.audit_log_load_timer, queries_deduped:
-            i = 0
+            last_log_time = datetime.now()
+            last_report_time = datetime.now()
             for _, query_instances in queries_deduped.items():
                 for query in query_instances.values():
-                    if i > 0 and i % 10000 == 0:
+                    now = datetime.now()
+                    if (now - last_log_time).total_seconds() >= 60:
```

Comment:
might be good to have a general abstraction for "print this message every X seconds", since the modulo stuff just feels like an approximation of that